modyn/utils/utils.py

```diff
@@ -123,3 +127,13 @@ def package_available_and_can_be_imported(package: str) -> bool:
 
 def flatten(non_flat_list: list[list[Any]]) -> list[Any]:
     return [item for sublist in non_flat_list for item in sublist]
+
+
+def seed_everything(seed: int) -> None:
+    random.seed(seed)
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
```

Comment:
I think we might also want to call this function on the selector when we instantiate the AbstractSelectionStrategy (in __init__) since some strategies use `random`

Reply:
Yep, there's a random shuffling of samples. I added a GRPC call from the supervisor to the selector just after the connection.